tensorly/decomposition/cmtf_als.py

```diff
@@ -0,0 +1,157 @@
+
+    return tl.min(FMS)
+
+
+def cmtf_als_for_third_order_tensor(tensor_3d, matrix, rank):
+    """
+    Calculates a coupled matrix and tensor factorization of 3rd order tensor and matrix which are coupled in first mode.
+
+    Assume you have tensor_3d = [[A, B, C]] and matrix = [[A, V]], which are coupled in 1st mode.
+    With coupled matrix and tensor factorization (CTMF), the factor matrices A, B, C for the CP decomposition of X
+    and the matrix V are found.
+    This implementation only works for a coupling in the first mode.
+
+    Solution is found via alternating least squares (ALS) as described in Figure 5 of
+    @article{acar2011all,
+      title={All-at-once optimization for coupled matrix and tensor factorizations},
+      author={Acar, Evrim and Kolda, Tamara G and Dunlavy, Daniel M},
+      journal={arXiv preprint arXiv:1105.3422},
+      year={2011}
+    }
+
+    Parameters
+    ----------
+    tensor_3d : tl.tensor or Kruskal tensor
+        3rd order tensor X = [[A, B, C]]
+    matrix : tl.tensor or Kruskal tensor
+        matrix that is coupled with tensor in first mode: Y = [[A, V]]
+    rank : int
+        rank for CP decomposition of X
+
+    Returns
+    -------
+    tensor_3d_pred, matrix_pred : Kruskal tensors
+        tensor_3d_pred = [[A,B,C]], matrix_pred = [[A,V]]
+
+    Examples
+    --------
+    A = tl.tensor([[1, 2], [3, 4]])
+    B = tl.tensor([[1, 0], [0, 2]])
+    C = tl.tensor([[2, 0], [0, 1]])
+    V = tl.tensor([[2, 0], [0, 1]])
+    R = 2
+
+    X = tl.kruskal_tensor.KruskalTensor((None, [A, B, C]))  # weights are None
+    Y = tl.kruskal_tensor.KruskalTensor((None, [A, V]))
+
+    tensor_3d_pred, matrix_pred = cmtf_als_for_third_order_tensor(X, Y, R)
+
+    """
+
+    # initialize values
+    shape_list = tensor_3d.shape
+    s = shape_list + (matrix.shape[1],)
+    A, B, C, V = random_kruskal(s, rank).factors
+
+    if isinstance(tensor_3d, tl.kruskal_tensor.KruskalTensor):
+        X = tl.kruskal_tensor.kruskal_to_tensor(tensor_3d)
+    else:
+        X = tensor_3d.copy()
+
+    if isinstance(matrix, tl.kruskal_tensor.KruskalTensor):
+        Y = tl.kruskal_tensor.kruskal_to_tensor(matrix)
+    else:
+        Y = matrix.copy()
+
+    objective_old = tl.tensor(np.inf)
+
+    # alternating least squares
+    # note that no rescaling is done since it is not guaranteed that the columns in true matrices have unit norm
+    # note that the order of the khatri rao product is reversed since tl.unfold has another order than assumed in paper
+    for iteration in range(10 ** 4):
+        A = solve_least_squares(tl.concatenate((khatri_rao([B, C]).T, V.T), axis=1).T,
+                                tl.concatenate((tl.unfold(X, 0), Y), axis=1).T).T
+        B = solve_least_squares(khatri_rao([A, C]), tl.unfold(X, 1).T).T
+        C = solve_least_squares(khatri_rao([A, B]), tl.unfold(X, 2).T).T
+        V = solve_least_squares(A, Y).T
+        objective_new = 1 / 2 * tl.norm(
```

Comment:
I guess this is actually the objective function, would error be more explicit? :/
No strong opinion though, just a comment :)